packages/mobile/src/verify/saga.ts

```diff
@@ -898,23 +887,28 @@ const sagas: Array<[string, any]> = [
   [revoke.type, revokeSaga],
 ]
 
+export function* watchVerificationEvents() {
+  for (const [actionType, saga] of sagas) {
+    yield takeEvery(actionType, callSagaAndHandleErrors(saga, actionType))
+  }
+}
+
 export function* verifySaga() {
+  yield spawn(watchVerificationEvents)
   while (true) {
-    const task: Task = yield fork(function* () {
-      Logger.debug(TAG, 'Verification Saga has started')
-      for (const [actionType, saga] of sagas) {
-        yield takeEvery(actionType, createErrorHandler(saga, actionType))
-      }
-    })
+    yield take(start.type)
+    Logger.debug(TAG, 'Verification Saga has started')
+
     const { cancelled, timedOut }: { cancelled: boolean; timedOut: boolean } = yield race({
       cancelled: take(cancel.type),
+      succeeded: take(succeed.type),
+      failed: take(fail.type),
```

Comment:
Should this also listen for `stop` and reset the timer on `reset`?

Reply:
For `stop` yes, added. For `reset` it's still technically part of the verification, it didn't end and resetting the timer  on it would be a bit tricky for the value-add so I'd vote for leaving it for now. We should definitely take another look at how all this works soon though